app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  belongs_to :user
+
+  def update_posts_counter
```

Comment:
- [ ] It's good practice if you put `update_posts_counter` after the private method. 
[More info...](https://guides.rubyonrails.org/active_record_callbacks.html#:~:text=before_validation%20%3Aensure_login_has_a_value-,private,-def%20ensure_login_has_a_value%0A%20%20%20%20%20%20if)